### ISSUE TITLE
mapping accurate short reads to high error rate long reads

### ISSUE THREAD

**Issue by leleory** [lh3/minimap2#214]
Hello there,
Can minmap2 map accurate short genomic reads to long PacBio or Nanopore reads/contigs?
If I have merged Nanopore reads (e.g. with miniasm or CANU, but error rate is still high because the coverage is low) would the below give accurate mapping (where ref.fa would be the contigs):
minimap2 -ax sr ref.fa reads-se.fq > aln.sam
or is this only recommended for mapping accurate reads to high quality genome assembly?
Thank you,
Lel

**Comment by lh3** [comment]
Possible duplicate: #90. In that issue, @sjackman decided not to map short reads to noisy long reads.
<hr/>
Canu contigs are pretty good. You can try the `sr` preset, perhaps with decreased `-w` trade performance for sensitivity. Something like

```sh
minimap2 -ax sr -w 5 ref.fa reads.fq
```

I don't think minimap2 will work well with uncorrected reads/contigs. If you have enough short reads, assemble them first and then map will be much better.

**Comment by leleory** [comment]
Hi Heng,
Thanks for the suggestion. The canu contigs are high error rate as the coverage was <5x. We are doing some additional sequencing to reach a min of 10x. I try to use nanopore ultra long reads to scaffold the assembled Illumina reads. Before that, though, I would like to error correct the contigs with the short reads.

I have tried both bwa and minimap2 to align the Illumina reads to the still "noisy" canu contigs using:
`minimap2 -ax sr -w 5 ref.fa read1.fq read2.fq`
`bwa mem -x ont2d`
Visually the alignments look OK. Is there a way I can assess mapping precision?

Mapping rate with bwa was 81% (65% in proper-pairs) with minimap2 it was 48% (18%).  Memory usage was 6Gb and 9Gb and speed was 104 CPUmin and 20 CPUmin per Million reads, respectively.
I do not have simulated reads to estimate mapping precision. Assuming that the SNP and indel counts are indicative to the sequencing errors I have counted the number of nucleotides within these categories. The SNPs and indels are 50% and 30% up in the bwa mapping. 

Thanks,
Lel

**Comment by tseemann** [comment]
@leleory if your genome is not too big, Unicycler is perfect for your data set:
https://github.com/rrwick/Unicycler

**Comment by leleory** [comment]
@tseemann: Thank you for the suggestion. It is a mammalian genome of 2.6Gb. I have tried the SPAdes-hybrid approach before and it failed. Unfortunately Unicycler also uses SPAdes, so it is possible that it will fail.
Ideally I would need a package which would be able to use the minimap2 based mapping between the ultra-long and Illumina contigs to create a merged assembly.
Any suggestion would be much appreciated.

**Comment by sjackman** [comment]
As far as I know, this pipeline does not yet exist. I can suggest using ABySS to assemble your genome, and then LINKS to scaffold with the Nanopore reads, both from Birol lab (my lab).
https://github.com/bcgsc/abyss
https://github.com/bcgsc/links
https://www.birollab.ca/software

**Comment by leleory** [comment]
Thank you all for the suggestions. While I managed to map the short reads with minimap2 to the ONT reads the overall coverage went above 200x (original data is 100x). I have looked into filtering based on MAPQ values to get a coverage close to the original 100x. 
In the end to avoid ambiguity due to the thresholding I followed Heng's suggestion, i.e. assembled the reads and mapped those to the long reads. 
@sjackman: thank you for pointing me to LINKS, I am looking into it.
